jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/ServerWebSocketContainer.java

```diff
@@ -81,9 +81,9 @@ public static ServerWebSocketContainer ensure(Server server, ContextHandler cont
                 ? WebSocketServerComponents.ensureWebSocketComponents(server)
                 : WebSocketServerComponents.ensureWebSocketComponents(server, contextHandler);
             WebSocketMappings mappings = new WebSocketMappings(components);
-            container = new ServerWebSocketContainer(mappings);
-            container.addBean(mappings);
-            context.setAttribute(WebSocketContainer.class.getName(), container);
```

Comment:
I think you should still set the attribute here.

Because now calling `ServerWebSocketContainer.ensure(...)` will return two different instances of `ServerWebSocketContainer`. This is because this attribute is not set until `doStart`, so  `ServerWebSocketContainer#get` will return null.

Reply:
This was determined to be safe, sane, and consistent to how the ws-container is available on jakarta websocket as well. (only available during webapp init)

Yes, the ServerWebSocketContainer.get() will not return unless the outer container is started.
We tested this with a CustomContextHandler that had a nested WebSocketUpgradeHandler.
The CustomContextHandler.doStart() can access use ServerWebSocketContainer.get() successfully.